aiida/cmdline/utils/common.py

```diff
@@ -410,3 +410,56 @@ def build_entries(ports):
     for exit_code in sorted(process_spec.exit_codes.values(), key=lambda exit_code: exit_code.status):
         message = exit_code.message.capitalize()
         click.secho('{:>{width_name}d}:  {}'.format(exit_code.status, message, width_name=max_width_name))
+
+
+def get_num_workers():  #pylint: disable=inconsistent-return-statements
+    """
+    Get the number of active daemon workers from the circus client
+    """
+    from circus.exc import CallError
+
+    from aiida.manage.manager import get_manager
+
+    manager = get_manager()
+    client = manager.get_daemon_client()
+
+    if client.is_daemon_running:
+        response = client.get_numprocesses()
+        if response['status'] != 'ok':
+            if response['status'] == client.DAEMON_ERROR_TIMEOUT:
+                raise CallError('verdi thought the daemon was alive, but the call to the daemon timed-out')
+            elif response['status'] == client.DAEMON_ERROR_NOT_RUNNING:
+                raise CallError('verdi thought the daemon was running, but really it is not')
+            else:
+                raise CallError
+        try:
+            return response['numprocesses']
+        except KeyError:
+            raise CallError('Circus did not return the number of daemon processes')
+
+
+def check_worker_load(active_slots):
+    """
+    Check if the percentage usage of the daemon worker slots exceeds a threshold.
+    If it does, print a warning.
+
+    The purpose of this check is to warn the user if they are close to running out of worker slots
+    which could lead to their processes becoming stuck indefinitely.
+
+    :param active_slots: the number of currently active worker slots
+    """
+    from aiida.cmdline.utils import echo
+    from aiida.manage.external.rmq import _RMQ_TASK_PREFETCH_COUNT
+
+    warning_threshold = 0.9  # 90%
+
+    slots_per_worker = _RMQ_TASK_PREFETCH_COUNT
+
+    active_workers = get_num_workers()
```

Comment:
This can potentially throw the `CallError`. It would be better to catch it here and print a `echo_critical` instead of letting the exception bubble up. That said, I would use an aiida internal exception to throw from `get_num_workers` instead of the circus one. Normally you would import those to be able to catch them, but if you are throwing them yourself, might as well use an internal exception